ocaml/xapi/cert_distrib.ml

```diff
@@ -399,6 +399,73 @@ let exchange_certificates_among_all_members ~__context =
     (fun host -> Worker.remote_regen_bundle host rpc session_id)
     all_hosts
 
+let ( (get_local_ca_certs : unit -> WireProtocol.certificate_file list)
+    , (get_local_pool_certs : unit -> WireProtocol.certificate_file list) ) =
+  let g path () =
+    Sys.readdir path
+    |> Array.to_list
+    |> List.map (fun filename ->
+           let path = Filename.concat path filename in
+           let content = string_of_file path in
+           WireProtocol.{filename; content}
+       )
+  in
+  (g ApplianceProvider.store_path, g HostPoolProvider.store_path)
+
+let am_i_missing_certs ~__context : bool =
+  (* compare what's in the database with what's on my filesystem *)
+  let missing_pool_certs () =
+    let expected_files =
+      Db.Host.get_all ~__context
+      |> List.map (fun self -> Db.Host.get_uuid ~__context ~self)
+      |> List.map (fun x -> x ^ ".pem")
+      |> StringSet.of_list
+    in
+    let actual_files =
+      Sys.readdir HostPoolProvider.store_path
+      |> Array.to_seq
+      |> StringSet.of_seq
+    in
+    StringSet.subset expected_files actual_files |> not
+  in
+  let missing_ca_certs () =
+    let expected_files =
+      Db.Certificate.get_all ~__context
+      |> List.filter_map (fun self ->
+             match Db.Certificate.get_type ~__context ~self with
+             | `ca ->
+                 Some (Db.Certificate.get_name ~__context ~self)
+             | _ ->
+                 None
+         )
+      |> List.map (fun x -> x ^ ".pem")
```

Comment:
I think this is wrong. CA cert records have .pem in their names. This would have been avoided by adding to the providers the transformation from id to filename. this is now done as part of the certificate_of_id_content.